miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java

```diff
@@ -505,6 +510,16 @@ private void loadChildEntities(Sample sample) throws IOException {
       if (sai.getSubproject() != null && sai.getSubproject().getId() != null) {
         sai.setSubproject(subProjectDao.getSubproject(sai.getSubproject().getId()));
       }
+      if (isTissueProcessingSample(sai) && sai instanceof SampleSlide) {
+        Stain originalStain = ((SampleSlide) sai).getStain();
+        Stain stain;
+        if (originalStain == null) {
+          stain = null;
+        } else {
+          stain = stainDao.get(originalStain.getId());
```

Comment:
I like the pattern where services have access to their own DAO and other services because it means we only need to implement things like filters for access in one place, instead of everywhere that might access that DAO (though that's certainly more relevant for samples than lists of stains). I think that pattern got messed up though when we were creating new services and didn't necessarily go back to pre-existing services to replace the DAO access with service access.